operations/host_spawn.go

```diff
@@ -681,6 +702,49 @@ func hostRunCommand() cli.Command {
 			client := conf.setupRestCommunicator(ctx)
 			defer client.Close()
 
+			var hostIDs []string
+			if hostID != "" {
+				hostIDs = []string{hostID}
+			} else {
+				var createdBeforeTime, createdAfterTime time.Time
+				if createdBefore != "" {
+					createdBeforeTime, err = time.Parse(time.RFC3339, createdBefore)
+					if err != nil {
+						return errors.Wrap(err, "can't parse created before time")
+					}
+				}
+				if createdAfter != "" {
+					createdAfterTime, err = time.Parse(time.RFC3339, createdAfter)
+					if err != nil {
+						return errors.Wrap(err, "can't parse create after time")
+					}
+				}
+
+				hosts, err := client.GetHosts(ctx, model.APIHostParams{
+					CreatedBefore: createdBeforeTime,
+					CreatedAfter:  createdAfterTime,
+					Distro:        distro,
+					UserSpawned:   userSpawned,
+					Mine:          mine,
+					Status:        evergreen.HostRunning,
+				})
+				if err != nil {
+					return errors.Wrapf(err, "can't get matching hosts")
+				}
+				if len(hosts) == 0 {
+					return errors.New("no matching hosts")
```

Comment:
this will cause the program to exit non-zero which might not be right

Reply:
fixed